src/sagas/projects.js

```diff
@@ -134,15 +138,30 @@ export function* projectExported({payload: {exportType}}) {
 }
 
 export function* saveCurrentProject() {
-  const userId = yield select(getCurrentUserId);
   const currentProject = yield select(getCurrentProject);
+  if (currentProject) {
+    yield* saveProjectWithKey(currentProject.projectKey);
+  }
+}
 
-  if (userId && currentProject && !isPristineProject(currentProject)) {
-    yield call(saveProject, userId, currentProject);
+function* saveProjectWithKey(projectKey) {
+  const state = yield select();
+  const userId = yield select(getCurrentUserId);
+  const project = yield call(getProject, state, {projectKey});
```

Comment:
It’s a little odd to call `select` twice, once with a selector and once for the whole state. Either of these would be fine:

```js
const userId = yield select(getCurrentUserId);
const project = yield select(partialRight(getProject, {projectKey}));
```

or

```js
const state = yield select();
const userId = yield call(getCurrentUserId, state);
const project = yield call(getProject, state, {projectKey});
```

I would say I have a mild preference for the first one but honestly doesn’t matter (and I probably wouldn’t even bring this up if it were the only feedback on the PR to be honest!)